src/routes/StackView/StackView.tsx

```diff
@@ -32,58 +53,156 @@ export const StackView = () => {
+      }
+      setTransitionConfig({
+        from: {
+          transform: getTransform("start", direction),
+          position: "absolute",
+          opacity: 0,
+        },
+        enter: {transform: "translate(0%)", position: "relative", opacity: 1},
+        leave: {
+          transform: getTransform("end", direction),
+          position: "absolute",
+          opacity: 0,
+        },
+        items: {
+          parent: note,
+          stack: stackedNotes,
+          avatar: author!.user.avatar,
+          authorName,
+        },
+      });
+      prevNote.current = note;
+    }
+  }, [author, authorName, columns, note, stackedNotes]);
 
   if (!note) {
     navigate(`/board/${boardId}`);
     return null;
   }
 
   const handleClose = () => {
-    if (moderating && (viewer.role === "MODERATOR" || viewer.role === "OWNER")) {
+    if (presenting) {
       dispatch(Actions.stopSharing());
     }
     navigate(`/board/${boardId}`);
   };
 
+  const handleModeration = (id: string) => {
+    if (presenting) {
+      dispatch(Actions.shareNote(id));
+    }
+  };
+
+  const navigationProps = {
+    stacks: stacksInColumn,
+    currentStack: note.id,
+    prevColumnStack: prevColumnParent,
+    nextColumnStack: nextColumnParent,
+    handleModeration,
+  };
+
   return (
     <Portal onClose={handleClose} className={classNames("stack-view__portal", getColorClassName(column!.color as Color))} hiddenOverflow centered disabledPadding>
       <div className={classNames("stack-view", getColorClassName(column!.color as Color))}>
         <NoteDialogComponents.Header columnName={column!.name} />
-        <NoteDialogComponents.Note
-          key={noteId}
-          noteId={noteId!}
-          text={note!.text}
-          authorId={note!.author}
-          avatar={author!.user.avatar}
-          authorName={authorName}
-          showAuthors={showAuthors}
-          onClose={handleClose}
-          onDeleteOfParent={handleClose}
-          showUnstackButton={false}
-          viewer={viewer}
-          className="stack-view__parent-note"
-        />
-        <NoteDialogComponents.Wrapper>
-          {stackedNotes.map((n) => (
-            <NoteDialogComponents.Note
-              key={n.id}
-              noteId={n.id}
-              text={n.text}
-              authorId={n.author}
-              avatar={n.avatar}
-              authorName={n.authorName}
-              showAuthors={showAuthors}
-              onClose={handleClose}
-              onDeleteOfParent={handleClose}
-              showUnstackButton
-              viewer={viewer}
-            />
-          ))}
-        </NoteDialogComponents.Wrapper>
+        <StackNavigation {...navigationProps} />
+        <div className="stack-view__content">
+          <Transition {...transitionConfig}>
+            {(styles: object, item: {parent: Note | undefined; stack: StackedNote[]; avatar: AvataaarProps | undefined; authorName: string}) => (
```

Comment:
I think a possible typing for the styles object would be the CSSProperties from the react module  (there might be more than this)

Reply:
and you can omit the `| undefined`:
`{parent?: Note; stack: StackedNote[]; avatar?: AvataaarProps; authorName: string}`